benchmark/src/bench_client.rs

```diff
@@ -90,16 +103,88 @@ impl BenchClient {
         &mut self,
         request: PutRequest,
     ) -> Result<PutResponse, BenchClientError> {
-        if self.use_curp_client {
-            let response = self.xline_client.kv_client().put(request).await?;
-            Ok(response)
-        } else {
-            let opts = from_request_to_option(&request);
-            let response = self
-                .etcd_client
-                .put(request.key(), request.value(), Some(opts))
-                .await?;
-            Ok(response.into())
+        match self.kv_client {
+            KVClient::Xline(ref mut xline_client) => {
+                let response = xline_client.kv_client().put(request).await?;
+                Ok(response)
+            }
+            KVClient::Etcd(ref mut etcd_client) => {
+                let opts = from_request_to_option(&request);
+                let response = etcd_client
+                    .put(request.key(), request.value(), Some(opts))
+                    .await?;
+                Ok(response.into())
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+#[allow(clippy::panic)]
+#[allow(clippy::unwrap_used)]
+#[allow(clippy::indexing_slicing)]
+mod test {
+    use crate::bench_client::{BenchClient, ClientOptions, KVClient, PutRequest};
+    use xline_client::types::kv::RangeRequest;
+    use xline_test_utils::Cluster;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_new_xline_client() {
+        // create xline client
+        let mut cluster = Cluster::new(3).await;
+        cluster.start().await;
+        let use_curp_client = true;
+        let config = ClientOptions::default();
+        let mut client = BenchClient::new(cluster.addrs(), use_curp_client, config)
+            .await
+            .unwrap();
+        //check xline clinet put value exist
+        let request = PutRequest::new("put", "123");
+        let _put_response = client.put(request).await;
+        match client.kv_client {
```

Comment:
Actually, you can implement a `get` method in `BenchClient` for testing only.

Reply:
Sounds good, I'll ping you again later after I've made the modifications.